### PR TITLE
Bug: H3 ::before pseudo-element is preventing click interation with the elements above

### PR DESCRIPTION
**Pull request by corwintines**
## Description
- Added scroll-margin-top to headers
- Added a hook in for when a page is loaded with a anchor tag and scroll to that anchor

## Related Issue
fixes: https://github.com/ethereum/ethereum-org-website/issues/4793

### CODE REVIEW

**Review by wackerow — CHANGES_REQUESTED** [review]
Lookin good

https://build-4d984deb-29ab-4739-bbb6-dae49915c8f4.gtsb.io/en/dao/#share-based-membership
Going here seems to properly navigate to the correct section with space for the nav bar, and the link above it remains clickable. 

Just left code suggestions to use our theme variable for the nav height instead of hard-coding.

**Review by wackerow — APPROVED** [review]

File: src/theme.js [hunk 1]
```diff
@@ -437,6 +435,7 @@ export const GlobalStyle = createGlobalStyle`
     line-height: 1.4;
     margin: 2rem 0;
     font-weight: 700;
+    scroll-margin-top: ${theme.variables.navHeight};
```
Comment (minimalsm) [inline src/theme.js:438]:
```suggestion
    scroll-snap-margin: ${theme.variables.navHeight};
    scroll-margin-top: ${theme.variables.navHeight};
```

**Review by minimalsm — COMMENTED** [review]
Nice @corwintines. Working well for me. 

Noticed on [caniuse.com](https://caniuse.com/?search=scroll-margin) that if we include polyfill with `scroll-snap-margin` we get an additional ~3% browser coverage for iOS devices. Probably worth it?

File: src/theme.js (hunk 1 above)
Comment (corwintines) [reply src/theme.js:438]: I tried using this and it wasn't working. Maybe we can make an issue to change over, but would like to see this get in for now if thats ok with you.

**Pull request merged**